File: android/app/src/main/java/com/cbq/nnlg/SemesterCourseList.java
package com.cbq.nnlg;

import java.util.ArrayList;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class SemesterCourseList {

    private String semesterCourseListHTML;

    private ArrayList<String> courseList = new ArrayList<>();

    private StringBuffer allJSON = new StringBuffer();



    public SemesterCourseList(String semesterCourseListHTML){

        this.semesterCourseListHTML = semesterCourseListHTML;

        getCourseList();
    }


    private void getCourseList(){

        Pattern pattern = Pattern.compile("学年学期：([\\w\\W]+)</select>");
        Matcher matcher = pattern.matcher(semesterCourseListHTML);

        if(matcher.find()){
            //System.out.println(matcher.group(1));

            Pattern pattern1 = Pattern.compile(">([\\s]*)([^<]{9,15})([\\s]*)</option>");
            Matcher matcher1 = pattern1.matcher(matcher.group(1));
            while(matcher1.find()){
                //System.out.println(matcher1.group(2));
                courseList.add(matcher1.group(2));
                //courseList.add()
            }

        }


    }



    public String getAllList(){

        allJSON.append("[");
        for(int i = 0 ; i < courseList.size() ; ++i ){




            allJSON.append("\""+ courseList.get(i) +"\"" );

            if(i!=courseList.size()-1)
                allJSON.append(",");

        }
        allJSON.append("]");

        return allJSON.toString();
    }



}

File: lib/utils/edusys/tools/SemesterCourseList.dart
/* FileName SemesterCourseList
 *
 * @Author 20840
 * @Date 2024/6/16 14:17
 *
 * @Description TODO
 */




class SemesterCourseList {

  String? _semesterCourseListHTML;

  List<String> courseList = [];

  String allJSON = "";



  SemesterCourseList(String semesterCourseListHTML){

    this._semesterCourseListHTML = semesterCourseListHTML;

    getCourseList();
  }


  void getCourseList(){
    RegExp regExp = RegExp(r'学年学期：([\w\W]+)</select>');
    RegExpMatch? match = regExp.firstMatch(_semesterCourseListHTML!);

    if(match!=null){
      RegExp regExp1 = RegExp(r'>([\s]*)([^<]{9,15})([\s]*)</option>');
      Iterable<Match> match1 = regExp1.allMatches(match.group(1).toString());
      for(Match m in match1){
        courseList.add(m.group(2).toString());
      }

    }


  }



  String getAllList(){

    allJSON+="[";
    for(int i = 0 ; i < courseList.length ; ++i ){




      allJSON+="\""+ courseList[i] +"\"";

      if(i!=courseList.length-1)
        allJSON+=",";

    }
    allJSON+="]";

    return allJSON.toString();
  }



}
